src/main/java/io/wispforest/owo/registration/reflect/FieldRegistrationHandler.java

```diff
@@ -88,4 +104,50 @@ private static <T> ReflectionUtils.FieldConsumer<T> createProcessor(ReflectionUt
         };
     }
 
+    private static <C, F> void iterateAccessibleStaticFieldsAllowingMemorizedSuppliers(Class<C> clazz, Class<F> targetFieldType, ReflectionUtils.FieldConsumer<F> fieldConsumer) {
+        for (var field : clazz.getDeclaredFields()) {
+            if (!Modifier.isStatic(field.getModifiers())) continue;
+
+            Object fieldValue;
+            try {
+                fieldValue = field.get(null);
+            } catch (IllegalAccessException e) {
+                continue;
+            }
+
+            if(fieldValue == null) continue;
+
+            var valueType = fieldValue.getClass();
+
+            F finalValue = null;
+
+            if (!targetFieldType.isAssignableFrom(valueType)) {
+                boolean isValid = false;
+
+                if(Supplier.class.isAssignableFrom(field.getType()) && field.getGenericType() instanceof ParameterizedType parameterizedType) {
+                    var genericType = parameterizedType.getActualTypeArguments()[0];
+
+                    if (genericType instanceof Class<?> genericClass && targetFieldType.isAssignableFrom(genericClass)) {
+                        if(!(fieldValue instanceof MemoizedEntry<?>)) {
+                            throw new IllegalStateException("A given Supplier object must be of a memoized variant or problems may occur! [Field: " + field.getName() + "]");
+                        }
+
+                        finalValue = (F) ((Supplier) fieldValue).get();
+
+
+                        isValid = true;
+                    }
+                }
+
+                if(!isValid) continue;
+            } else {
+                finalValue = (F) fieldValue;
+            }
+
+            if (field.isAnnotationPresent(IterationIgnored.class)) continue;
```

Comment:
why is this check after the checks for the target field type and for the supplier?